cmd2.py

```diff
@@ -340,6 +365,14 @@ def parse_quoted_string(cmdline):
     return lexed_arglist
 
 
+def with_category(category):
```

Comment:
I like that you added a decorator, that should make the code clean and easy to read.

I strongly encourage you to make one simple change to your decorator to help avoid order-of-operations dependencies when multiple decorators are involved.

You should apply the "@functools.wraps(func)" decorator to your internal **cat_decorator** function - see how we do it in other decorators below such as **with_argument_list**:
```Python
def with_category(category):
    """A decorator to apply a category to a command function"""
    @functools.wraps(func)
    def cat_decorator(func):
        categorize(func, category)
        return func
    return cat_decorator
```

Reply:
I don't think that's necessary in this situation because this decorator function is truly a decorator function rather than a wrapping function. This adds an attribute the the passed-in function and returns it.

Reply:
You make a good point, your decorator is mutating and returning the original function.  I think you are fine as-is.